src/3d/Options.fst

```diff
@@ -9,33 +9,80 @@ let input_file : ref (list string) = alloc []
 let error_log : ref (option string) = alloc None
 let error_log_function : ref (option string) = alloc None
 let debug : ref bool = alloc false
+let batch : ref bool = alloc false
+let clang_format : ref bool = alloc false
+let clang_format_executable : ref string = alloc ""
+let cleanup : ref bool = alloc false
 
-let options =
+(* We would like to parse --help as an option, but this would
+   require to recurse on the definition of the list of options. To
+   avoid that, we define this list as mutable, and then we add --help
+   after the fact. *)
+
+let options0 =
   let open FStar.Getopt in
   [(noshort, "module_name", OneArg ((fun mname -> module_name := Some mname), "module name"), "module name to use for the output file");

```

Comment:
Wonder if we should just ditch the module_name option and be uniform about using the file name prefix as the module name. We could do that in a separate PR.

Reply:
Yes, we could. What was the initial intent behind `module_name`?

Reply:
No good reason, I think. Not sure why I didn't go for the file name thing to begin with.